lib/instrumentation/openai.js

```diff
@@ -23,6 +23,69 @@ const { DESTINATIONS } = require('../config/attribute-filter')
 
 let TRACKING_METRIC = OPENAI.TRACKING_PREFIX
 
+/**
+ * Parses the response from OpenAI and extracts the message content and role.
+ *
+ * @param {object} response The OpenAI SDK response object
+ * @returns {{ content: string, role: string }} the message object with fields `content` and `role`
+ */
+function getMessageFromResponse(response) {
+  let content
+  let role
+  if (response?.object === 'response') {
+    content = response?.output?.[0]?.content?.[0]?.text
+    role = response?.output?.[0]?.role
+  } else {
+    content = response?.choices?.[0]?.message?.content
+    role = response?.choices?.[0]?.message?.role
+  }
+
+  return { content, role }
+}
+
+/**
+ * Parses all messages from the OpenAI request object.
+ *
+ * @param {object} request The OpenAI SDK request object
+ * @param {Shim} shim instance of shim
+ * @returns {Array<object>} an array of message objects with fields `content` and `role`
+ */
+function getMessagesFromRequest(request, shim) {
+  // There are a few different ways to pass messages to OpenAI SDK.
+  //
+  // For langchain and `chat.completions.create`, messages are passed
+  // as an array of objects with `content` and `role` properties
+  // to the `messages` field of the request.
+  //
+  // For `responses.create`, messages are passed as an array of objects
+  // with `content` and `role` properties OR as a single string (implied
+  // to be a user message) to the `input` field of the request.
+
+  if (Array.isArray(request?.input)) {
+    return request.input
+      .filter(msg => msg?.content && msg?.role)
+      .map(msg => ({
+        content: msg.content,
+        role: msg.role
+      }))
+  } else if (typeof request?.input === 'string') {
+    return [{
+      content: request.input,
+      role: 'user'
+    }]
+  } else if (Array.isArray(request?.messages)) {
+    return request.messages
+      .filter(msg => msg?.content && msg?.role)
+      .map(msg => ({
+        content: msg.content,
+        role: msg.role
+      }))
+  } else {
+    shim.logger.warn('No valid messages found in OpenAI request object.')
+    return []
```

Comment:
If every condition in a conditional statement results in a `return`, and the last part of the statement is an `else`, that could be refactored to have the last `return` in the outer scope. (See [this post](https://medium.com/@rachelchervin/if-else-vs-if-return-using-implicit-returns-in-javascript-and-react-bbac63656e1b) for an example.) Alternatively, you could use the conditional to assign values to a a particular variable that's returned by the function.